programmers/두-정수-사이의-합.js

```diff
@@ -0,0 +1,14 @@
+function solution(a, b) {
+    let sum = 0;
+    
+    if ( a >= b) {
+        for ( let i = a; i >= b; i--) {
+            sum += i;
+        }    
+    } else if ( a < b) {
```

Comment:
```suggestion
    } else {
```
선택사항이 2개이기 때문에 else만 써줘도 괜찮아요ㅎㅎ

Reply:
```
    const MAX_NUM = Math.max(a, b);
    const MIN_NUM = Math.min(a, b);
    let sum = 0;
    
    for (let i = MIN_NUM; i <= MAX_NUM; i++) {

i <= MAX_NUM 과 i < MAX_NUM + 1 중 어느걸 사용하는게 더 깔끔한건가요??
```

Reply:
>     const MAX_NUM = Math.max(a, b);
>     const MIN_NUM = Math.min(a, b);
>     let sum = 0;
>     
>     for (let i = MIN_NUM; i <= MAX_NUM; i++) {
> 
> i <= MAX_NUM 과 i < MAX_NUM + 1 중 어느걸 사용하는게 더 깔끔한건가요??

설명드리기 앞서 일반적인 범위 표현 방식에 대해 말씀드릴게요.
For문 등 범위를 표현할 때는 00 이상 00 미만(ex: let i = 0; i < 배열의 길이; i++;) 으로 표현하는 것. 즉, 반열린 구간을 사용하는 것이 좋습니다.

* 닫힌구간(<= i <=)
* 열린구간( < i < )
* 반 열린 구간( <= i < )

대부분의 프로그래밍 언어가 이런식으로 표현하고 이는 다음과 같이 세가지 장점이 있음
     1. 공집합을 표현가능 ( 2<=i<2 ) 
     2. 두 구간이 연속한지 확인가능 ( [a,b), [c,d) ) 에서 b=c가 입증되면 a,d는 연속 
     3. 구간의 크기 쉽게 확인가능. b-a가 구간의 크기
     4. 모든 범위 표현을 하나로 통일함으로써 코드를 해석하는 것이 수월해짐

* 2번에서 [a, b) 의 형식은 일반적으로 프로그래밍의 범위 표현 방식 중 반열린 구간을 의미. 즉, a이상 b미만.
* 자바스크립트의 slice 메서드만 보더라도 반열린구간으로 배열을 잘라서 반환하는 것을 알 수 있습니다. 프로그래밍에서는 거의 모든 범위를 반열린구간으로 표현합니다.

For 등에서 i = 1부터 시작하는 것보다 0으로 시작하는 것이 좋은 이유는 
   1. 메모리 구조상(0번지로 시작한다)
   2. 반열린구간을 사용할 수 있다.

이 정도로 이해하시면 됩니다.

이제 질문하신 것에 대한 답변 드릴게요.
> i <= MAX_NUM 과 i < MAX_NUM + 1 중 어느걸 사용하는게 더 깔끔한건가요??

반열린구간을 사용하는 것이 좋다지만 상황에 따라 유두리 있게 사용할 줄 알아야합니다. 
위와 같은 경우는 반열린구간을 사용한다면 굳이 +1을 해주므로 코드해석에 오히려 방해요소가 될 수 있죠.
따라서 닫힌구간이 가독성에 좋다할 수 있습니다.

이는 개발자의 성향마다 다릅니다.
누구는 반열린구간을 써라.
누구는 열린구간을 써라.
아직도 명확한 답은 나와있지않죠.

본인이 재직하는 환경에서 요구하는 것에 따르는 것이 정답입니다.

저의 성향은 위와 같을 땐 “열린구간을 사용하는 것이 가독성에 좋다” 입니다.

창훈님 생각도 듣고싶네요🙏🏻
답변이 되었기를!